src/docfx/lib/CachedTokenCredential.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
We probably don't need this, its OK to cache the token once per process per environment using `Lazy<T>` just like before.